test-js/unit/pentaho/visual/base/View.Spec.js

```diff
@@ -34,155 +33,341 @@ define([
+      it("should throw if invoked with no arguments", function() {
         expect(function() {
           return new View();
-        }).toThrow(errorMatch.argRequired("element"));
-
-        expect(function() {
-          return new View(element);
         }).toThrow(errorMatch.argRequired("model"));
 
         expect(function() {
-          return new View(element, model);
+          return new View(model);
         }).not.toThrow();
       });
 
-      it("should throw if the first argument is not a DOM element", function() {
-        [
-          "div", 1, true, {}, []
-        ].forEach(function(elem) {
-          expect(function() {
-            return new View(elem, model);
-          }).toThrow(errorMatch.argInvalidType("element", "HTMLElement", typeof elem));
-        });
-      });
-
     });
 
     describe("validation: ", function() {
 
       it("should be valid if the model is valid", function(){
-        var view = new View(element, model);
+        var view = new View(model);
         expect(model.validate()).toBeNull(); //Null === no errors
         expect(view._isValid()).toBe(true);
       });
 
       it("should be invalid if the model is invalid", function(){
         var model = new Model();
-        var view = new View(element, model);
+        var view = new View(model);
         expect(model.validate()).not.toBeNull(); //Null === no errors
         expect(view._isValid()).toBe(false);
       });
+    });
+
+    describe("#update", function() {
+      var view;
+
+      var DerivedView = View.extend({
+        _update: function(){ return "Rendered"; }
+      });
+
+      var ValidationErrorView = DerivedView.extend({
+        _validate: function(){ return ["Some error"]; }
+      });
+
+      var UpdateErrorView = View.extend({
+        _update: function(){ throw new Error("Some error"); }
+      });
+
+      beforeEach(function() {
+        listeners = jasmine.createSpyObj('listeners', [
+          'didCreate',
+          'willUpdate',
+          'didUpdate',
+          'rejectedUpdate'
+        ]);
+
+        model.on("will:update", listeners.willUpdate);
+        model.on("did:update", listeners.didUpdate);
+        model.on("rejected:update", listeners.rejectedUpdate);
+
+        model.on("did:create", listeners.didCreate);
+      });
+
+      function expectValidationError(errorMessage, expectedMessage) {
+        expect("View update was rejected:\n - " + expectedMessage).toBe(errorMessage);
+      }
+
+      it("should call the `will:update` event listener if the view is valid", function(done) {
+        view = new DerivedView(model);
+
+        view.update().then(function resolved() {
+          expect(listeners.willUpdate).toHaveBeenCalled();
+          done();
+        }, function rejected() {
+          done.fail();
+        });
+      });
+
+      it("should call the `did:update` event listener if the view is valid and called `_update",
```

Comment:
Please use the asynchronous tests pattern provided by the "tests/test-utils". See `Context.Spec.js`.